### ISSUE TITLE
END event gets emitted multiple times

### ISSUE THREAD

**Issue by Vearmox** [chill117/proxy-lists#102]
```
        const proxyList = require('proxy-lists');
        const options = {
            countries: ['de']
        };

        const gettingProxies = proxyList.getProxiesFromSource('proxylists-net', options);
        const untestedProxies = [];

        gettingProxies.on('data', function(proxies) {
            untestedProxies.push(proxies);
        });

        gettingProxies.on('error', function (error) {
            console.error(error);
        });

        gettingProxies.once('end', function () {
            console.log('test');
        });

```
Excecuting this code results in the following terminal output:
```test 
test 
test  
test  
test
test
test
test
test
```

**Comment by chill117** [comment]
A lot of work was done to improve error handling and to ensure that the end event for each source is emitted properly. These changes have now been included in the latest release (v1.19.0)

Can you please retry your test with the latest version?

**Comment by chill117** [comment]
I've run the code cited above and I see only one "test" output to the console. So I will mark this as closed.
